modules/ssh/init.zsh

```diff
@@ -6,10 +6,15 @@
 #
 
 # Return if requirements are not found.
-if [[ "$OSTYPE" == darwin* ]] || (( ! $+commands[ssh-agent] )); then
-  return 1
+if [[ "$OSTYPE" == darwin* ]]; then
+    if [[ "$(which ssh-agent)" == "/usr/bin/ssh-agent" ]]; then
```

Comment:
The commands array lets me check the existence of a program, but it doesn't allow me to know where it's executable is located.  On OS X the system-provided ssh-agent is in /usr/bin/ssh-agent.  If you use Fink or MacPorts it's either in /opt or in /usr/local, if you use Homebrew it's in /usr/local.

Why it's important to differentiate is that the system-provided ssh-agent has a set of Apple custom patches to have non-standard behavior.  This behavior is a combination of support for integration with the OS X keychain (which used to be patchable to upstream via Homebrew but is no longer) and some additional flags it supports which are expected by how launchd interacts with it.

When using the system provides ssh-agent, rather than it running in the background, when a call is made to its socket, launchd starts it temporarily to respond to the request using socket redirection and then once the request is completed it shuts the agent down again.  The agent with Apple's patches is just a wrapper for the OS X keychain.

However, if you want something newer than the ancient version provided by OS X, you need to install it from Fink, MacPorts, Homebrew, or manually.  Once you've done so, it's behavior matches the expected behavior for all other POSIX systems.  In the original case, it was checking to see if the system was OS X or lacked ssh-agent, if either was true it merely returned 1 and then exited.  In my patch, it now checks to see if your OS is OS X, and if so, checks to see if you're using the system provided ssh-agent, if so it returns 1, otherwise it continues.  If you don't have ssh-agent installed, it returns 1, otherwise it continues.  This logic flow allows it to work properly on both modified and unmodified OS X systems.